cmd/demo/node.go

```diff
@@ -258,15 +260,29 @@ func (n *node) HandleProposal(prop client.ChannelProposal, res *client.ProposalR
 	}
 	n.log.WithField("peer", id).Debug("Channel propsal")
 
-	fmt.Printf("\n💭 Received channel proposal from %v with funding %v.\n", alias, weiToEther(req.InitBals.Balances[0]...))
-
-	a := req.Accept(n.offChain.Address(), client.WithNonce(nonce()))
-	if _, err := res.Accept(ctx, a); err != nil {
-		n.log.Error(errors.WithMessage(err, "accepting channel proposal"))
-		return
-	}
-
-	fmt.Println("✅ Channel proposal accepted")
+	bals := weiToEther(req.InitBals.Balances[0]...)
+	theirBal := bals[0] // proposer has index 0
+	ourBal := bals[1]   // proposal receiver has index 1
+	msg := fmt.Sprintf("🔁 Incoming channel proposal from %v with funding [My: %v Ξ, Peer: %v Ξ].\nAccept (y/n)? ", alias, ourBal, theirBal)
+	Prompt(msg, func(userInput string) {
+		ctx, cancel := context.WithTimeout(context.Background(), config.Node.HandleTimeout)
+		defer cancel()
+
+		if userInput == "y" {
+			fmt.Printf("✅ Channel proposal accepted. Opening channel...\n")
+			a := req.Accept(n.offChain.Address(), client.WithNonce(nonce()))
+			if _, err := res.Accept(ctx, a); err != nil {
+				n.log.Error(errors.WithMessage(err, "accepting channel proposal"))
+				return
```

Comment:
you are right, currently the functionality would not change if we remove `return`. however, logically, the function should always return in case of error and I like to be explicit about that. for example, if some other develop later adds code below and is not aware that the function already failed, this could be problematic.